libbeat/feature/bundle.go

```diff
@@ -38,7 +38,7 @@ func (b *Bundle) Filter(stabilities ...Stability) *Bundle {
 
 	for _, feature := range b.features {
 		for _, stability := range stabilities {
-			if feature.Stability() == stability {
+			if feature.Description().Stability() == stability {
```

Comment:
How about updating the function to accept a predicate.

e.g.

```
func (b *Bundle) Filter(pred func(Describer) bool) *Bundle {
  var filtered []Featurable
  for _, feature := range b.features {
    if pred(feature) {
      filtered = append(filtered, feature)
    }
  }
  return NewBundle(filtered...)
}
```

Filtering for stability can be provided using a predicat constructor:

```
func HasStabilityPred(stabilities ...Stability) func(Describer) bool {
  return func(d Describer) bool {
}
```

Reply:
Im OK with this change, Not sure why this pattern never come to my mind.

Reply:
I like the idea, the following chain will be possible.

```golang
bundle.FilterWith(NamespaceMatchPred("processor"))
   .FilterWith(HasStabilityPred(feature.Beta))
```